src/user-management/interfaces/authentication-events.interface.ts

```diff
@@ -0,0 +1,27 @@
+type EventStatus =
+  | {
+      status: 'succeeded';
+    }
+  | {
+      status: 'failed';
+      error: {
+        code: string;
+        message: string;
+      };
+    };
+
+export type EmailVerification = {
```

Comment:
I'm thinking of using this same module for other authentication event types, such as Magic Auth, MFA, etc, they share a similar shape, and create something like `magic-auth.interface.ts` looks like an entity when it's actually not 🤔

Reply:
I could also create `email-verification.interface.ts` but with `EmailVerificationEvent` as the interface exported from there, I think this would be better.

Reply:
what about one shared `AuthenticationEvent` type?

Reply:
That sounds better to me! Going to update the code.